scripts/lib.sh

```diff
@@ -146,25 +146,59 @@ function show_verbose_output()
 
 function await_gateway_registered() {
     until [ "$($FM_MINT_CLIENT list-gateways | jq -e ".num_gateways")" = "1" ]; do
-        sleep $POLL_INTERVAL
+        sleep $FM_POLL_INTERVAL
     done
 }
 
+function run_dkg() {
+  # Generate federation configs
+  BASE_PORT=$((8173 + 10000))
+  CERTS=""
+  for ((ID=0; ID<FM_FED_SIZE; ID++));
+  do
+    echo "making dir"
+    mkdir $FM_CFG_DIR/server-$ID
+    FED_PORT=$(echo "$BASE_PORT + $ID * 10" | bc -l)
+    API_PORT=$(echo "$BASE_PORT + $ID * 10 + 1" | bc -l)
+    export FM_PASSWORD="pass$ID"
+    echo "making creating cert for port $FED_PORT $API_PORT"
+    RUST_BACKTRACE=1 $FM_BIN_DIR/distributedgen create-cert --p2p-url ws://127.0.0.1:$FED_PORT --api-url ws://127.0.0.1:$API_PORT --out-dir $FM_CFG_DIR/server-$ID --name "Server-$ID"
+    CERTS="$CERTS,$(cat $FM_CFG_DIR/server-$ID/tls-cert)"
+  done
+  CERTS=${CERTS:1}
+  echo "Running DKG with certs: $CERTS"
+
+  DKG_PIDS=""
+  for ((ID=0; ID<FM_FED_SIZE; ID++));
+  do
+    export FM_PASSWORD="pass$ID"
+    fed_port=$(echo "$BASE_PORT + $ID * 10" | bc -l)
+    api_port=$(echo "$BASE_PORT + $ID * 10 + 1" | bc -l)
+    $FM_BIN_DIR/distributedgen run  --bind-p2p 127.0.0.1:$fed_port --bind-api 127.0.0.1:$api_port --out-dir $FM_CFG_DIR/server-$ID --certs $CERTS &
+    DKG_PIDS="$DKG_PIDS $!"
+  done
+  wait $DKG_PIDS
```

Comment:
Now we can't just `wait` for all we've processes spawned, so instead we keep track of just the DKG PIDs and only wait for them.